controllers/openstackdataplane_controller.go

```diff
@@ -283,18 +308,21 @@ func (r *OpenStackDataPlaneReconciler) CreateDataPlaneNode(ctx context.Context,
 	return nil
 }
 
-// CreateDataPlaneRole -
-func (r *OpenStackDataPlaneReconciler) CreateDataPlaneRole(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlane, helper *helper.Helper) error {
-
+// CreateOrPatchDataPlaneRoles Create or Patch DataPlaneRole
+func CreateOrPatchDataPlaneRoles(ctx context.Context,
+	instance *dataplanev1beta1.OpenStackDataPlane, helper *helper.Helper,
+	roleManagedHostMap map[string]map[string]string) error {
+	client := helper.GetClient()
+	logger := helper.GetLogger()
```

Comment:
Curious to understand why we're changing this from a receiver function? That's what allows us to do `r.Log` and `r.Client` without reinitializing those objects here with `helper.Get*()`.  

There might be some technical reason that I'm not considering, but it could also be that the functions were only recently changed. Maybe the change here is unintentional, so asking just in case.

Reply:
Hmm.. I guess you're aware that I've been working on this patch before your changes merged and this was rebased automatically.

Also, I mentioned that we either use the helper or use public properties from reconciler using the receiver. There is no point doing both. You can use r.Scheme rather